pkg/recovery/repair_test.go

```diff
@@ -119,41 +233,66 @@ func TestRecoveryHookCalls(t *testing.T) {
 
 // TestNewRepairHandler tests the function of repairing a chunk when a request for chunk repair is received.
 func TestNewRepairHandler(t *testing.T) {
-	logger := logging.New(ioutil.Discard, 0)
-
 	t.Run("repair-chunk", func(t *testing.T) {
 		// generate test chunk, store and publisher
-		c1 := chunktesting.GenerateTestRandomChunk()
+		payload := make([]byte, swarm.ChunkSize)
+		_, err := rand.Read(payload)
+		if err != nil {
+			t.Fatal(err)
+		}
+		c1, err := content.NewChunk(payload)
+		if err != nil {
+			t.Fatal(err)
+		}
 
 		// create a mock storer and put a chunk that will be repaired
 		mockStorer := storemock.NewStorer()
 		defer mockStorer.Close()
-		_, err := mockStorer.Put(context.Background(), storage.ModePutRequest, c1)
+		_, err = mockStorer.Put(context.Background(), storage.ModePutRequest, c1)
 		if err != nil {
 			t.Fatal(err)
 		}
 
 		// create a mock pushsync service to push the chunk to its destination
 		var receipt *pushsync.Receipt
+		var socChunk swarm.Chunk
 		pushSyncService := pushsyncmock.New(func(ctx context.Context, chunk swarm.Chunk) (*pushsync.Receipt, error) {
-			receipt = &pushsync.Receipt{
-				Address: swarm.NewAddress(chunk.Address().Bytes()),
+			if receipt == nil {
```

Comment:
This tests the RepairHandler, and if that functions properly the message is sent to push sync. So push sync and receipt is mocked here

Reply:
you  should not test stuff that you explicitly extend the test  with

Reply:
it is not tested.. since the pushsync is called twice... this mockpushsyncserice is handling both the case..
in the first case it should send the receipt, in the second case it (if the receipt is sent), it should send the receovery response.